server/src/server.ts

```diff
@@ -91,7 +99,39 @@ connection.onCompletion((textDocumentPosition: TextDocumentPositionParams): Comp
   if (documentAsText === undefined) {
     return []
   }
-  return contextHandler.getComletionItems(textDocumentPosition, documentAsText)
+
+  const word = analyzer.wordAtPointFromTextPosition({
+    ...textDocumentPosition,
+    position: {
+      line: textDocumentPosition.position.line,
+      // Go one character back to get completion on the current word. This is used as a parameter in descendantForPosition()
+      character: Math.max(textDocumentPosition.position.character - 1, 0),
+    },
+  })
+
+  logger.debug(`onCompletion - current word: ${word}`)
+
+  let symbolCompletions: CompletionItem[] = []
+  if(word){
+    const symbols = analyzer.getGlobalDeclarationSymbols(textDocumentPosition.textDocument.uri)
+
+    //Covert symbols to completion items
+    //TODO: remove duplicate symbols
+    symbolCompletions = symbols.map((symbol: SymbolInformation) => (
+      {
+        label: symbol.name,
+        kind: symbolKindToCompletionKind(symbol.kind),
+        documentation: `${symbol.name}`
+      }
+    ))
+  }
+  
+  const allCompletions = [
+    ...contextHandler.getComletionItems(textDocumentPosition, documentAsText),

```

Comment:
I don't think the old "getComletionItems" works. Don't hesitate to get rid of it if it is the case

Reply:
It actually works, when you declare a variable and type its name below, you will see two completion items. One is from my code and the other is from the `contextHandler`. 
I am planning to get rid of it later when I fully understand what it was doing.